src/main/java/forestry/core/genetics/GenomeSaveHandler.java

```diff
@@ -0,0 +1,82 @@
+package forestry.core.genetics;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+
+import forestry.api.genetics.AlleleManager;
+import forestry.api.genetics.IChromosome;
+import forestry.api.genetics.IChromosomeType;
+import forestry.api.genetics.IGenome;
+import forestry.api.genetics.IIndividual;
+import forestry.api.genetics.ISpeciesRoot;
+
+public class GenomeSaveHandler {
+
+	private  static SaveFormat writeFormat = SaveFormat.UID;
+
+	public  static void setWriteFormat(SaveFormat writeFormat) {
+		GenomeSaveHandler.writeFormat = writeFormat;
+	}
+
+	public static SaveFormat getFormat(NBTTagCompound tagCompound){
+		for(SaveFormat format : SaveFormat.values()){
+			if(format.canLoad(tagCompound)){
+				return format;
+			}
+		}
+		return SaveFormat.UID;
+	}
+
+	public static NBTTagCompound writeTag(IGenome genome, NBTTagCompound tagCompound) {
+		return writeFormat.writeTag(genome, tagCompound);
+	}
+
+	public static IChromosome[] readTag(NBTTagCompound tagCompound, ISpeciesRoot speciesRoot) {
+		SaveFormat format = getFormat(tagCompound);
+		return format.readTag(tagCompound, speciesRoot);
+	}
+
+	@Nullable
+	public static IChromosome getChromosomeDirectly(NBTTagCompound genomeNBT, IChromosomeType chromosomeType) {
```

Comment:
We need `getAlleleDirectly(NBTTagCompound genomeNBT, IChromosomeType chromosomeType, boolean active)` instead. 
Ideally we should only read 1 nbt tag to find the species for rendering. Right now your `Genome.getAlleleDirectly` is loading a chromosome which requires reading 2 tags, one for each allele.

Reply:
Do we not need the whole chromosome to find out witch of the two alleles is the active / dominant one ?

Reply:
We should store the active alleles as the first in each pair.

Reply:
That means we check at the creation of the chromosome which of the two alleles is the active one ?

Reply:
I'm not sure the order matters. For picking which one is used we choose the active one of the pair, and for breeding we pick one randomly. I can't think of a real relation between the alleles in a chromosome, besides that they came from one parent. I think it should be ok to mix them up unless you see a reason not to.